scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,1009 @@
+            TaskLogger.getInstance().info(taskId, channelHeader + "New channel content : " + getChannel(channel));
+        }
+    }
+
+    @Override
+    public void runActivity(Body body) {
+        Service service = new Service(body);
+        while (body.isActive()) {
+            try {
+
+                NewRequestWithWaitTime requestWithWaitTime = waitForNewRequest(service);
+                Request request = requestWithWaitTime.getNewRequest();
+
+                if (request != null && request.getMethodName().startsWith("wait") &&
+                    !testWaitFunction(service, request)) {
+                    // If the predicate is not met, delay the wait method execution
+                    TimedOutRequest timedOutRequest = new TimedOutRequest(request,
+                                                                          extractWaitRequestTimeoutParameter(request));
+                    logger.trace("New pending wait request : " + timedOutRequest);
+                    waitUntilQueue.add(new TimedOutRequest(request, extractWaitRequestTimeoutParameter(request)));
+                } else if (request != null) {
+                    service.serve(request);
+                }
+                unblockWaitMethods(service, requestWithWaitTime.getWaitTime());
+
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+            }
+        }
+    }
+
+    /**
+     * Wait for a new request in the request queue, until a request is found or until a timeout based on the pending wait request queue
+     * @param service active object service
+     * @return composite object containing the request and the waited time
+     * @throws InterruptedException
+     */
+    private NewRequestWithWaitTime waitForNewRequest(Service service) throws InterruptedException {
+        // we cannot wait more then min(pending_wait_requests_timeouts)
+        long maximumTimeToWaitForNewRequests = computeMinimumTimeout();
+
+        logger.trace("Waiting for new request with timeout = " + maximumTimeToWaitForNewRequests + " ms");
+        long timeSpentWaiting = System.currentTimeMillis();
+        Request request = service.blockingRemoveOldest(maximumTimeToWaitForNewRequests);
+        timeSpentWaiting = System.currentTimeMillis() - timeSpentWaiting;
+        logger.trace("Time spent waiting for new request: " + timeSpentWaiting + " ms");
+
+        return new NewRequestWithWaitTime(request, timeSpentWaiting);
+    }
+
+    @java.lang.SuppressWarnings("squid:S1481")
+    private long extractWaitRequestTimeoutParameter(Request request) {
+        // Extract all common parameters to waitUntil waitUntilThen
+        int paramIndex = 0;
+        // The following code is mainly for clarity
+        String originator = (String) request.getParameter(paramIndex++);
+        TaskId taskId = (TaskId) request.getParameter(paramIndex++);
+        String channel = (String) request.getParameter(paramIndex++);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "channel". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)